### PR TITLE
Text-decoration property in alfa-css

### PR DESCRIPTION
**Pull request by NiclasHedam**

### CODE REVIEW

**Comment by NiclasHedam** [comment]
There are also a few additional properties defined in the Level 4 draft. I've chosen to look at them at a later time since multiple of the properties has a grammar related to width, that can be reused when width is implemented.

File: packages/alfa-css/src/grammars/text-decoration-style.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  TextDecorationStyle,
+  T
+>;
+
+const ident: Production<Ident> = {
+  token: TokenType.Ident,
+  prefix(token) {
+    switch (token.value) {
+      case "initial":
+        return "none";
```
Comment (kasperisager) [inline packages/alfa-css/src/grammars/text-decoration-style.ts]: Everything `initial` and `inherit` is handled centrally.

File: packages/alfa-css/src/properties/text-decoration-style.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    return parse(input, TextDecorationStyleGrammar);
+  },
+  initial() {
+    return "none";
+  },
+  computed(getProperty, getParentProperty) {
+    const value = getProperty("textDecorationStyle");
+    if (value === "inherit") {
+      return getParentProperty("textDecorationStyle");
+    }
```
Comment (kasperisager) [inline packages/alfa-css/src/properties/text-decoration-style.ts]: Same goes for this; `inherit` and `initial` and handled centrally.

File: packages/alfa-css/test/grammars/text-decoration-style.spec.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+  t.deepEqual(
+    parse(lex(input, Alphabet), TextDecorationStyleGrammar),
+    expected,
+    input
+  );
+}
+
+test("Can parse initial text-decoration-style", t => {
+  textDecorationStyle(t, "initial", null);
+});
```
Comment (kasperisager) [inline packages/alfa-css/test/grammars/text-decoration-style.spec.ts]: This test case can be left out.

File: packages/alfa-css/test/grammars/text-decoration-style.spec.ts [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+  expected: TextDecorationStyle | null
+) {
+  t.deepEqual(
+    parse(lex(input, Alphabet), TextDecorationStyleGrammar),
+    expected,
+    input
+  );
+}
+
+test("Can parse initial text-decoration-style", t => {
```
Comment (kasperisager) [inline packages/alfa-css/test/grammars/text-decoration-style.spec.ts]: `initial` -> `none`?

File: packages/alfa-css/test/grammars/text-decoration-line.spec.ts [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+  expected: TextDecorationLine | null
+) {
+  t.deepEqual(
+    parse(lex(input, Alphabet), TextDecorationLineGrammar),
+    expected,
+    input
+  );
+}
+
+test("Can parse initial text-decoration-line", t => {
```
Comment (kasperisager) [inline packages/alfa-css/test/grammars/text-decoration-line.spec.ts]: `initial` -> `none`?

Comment (NiclasHedam) [reply packages/alfa-css/test/grammars/text-decoration-line.spec.ts]: Yup

File: packages/alfa-css/test/grammars/text-decoration-style.spec.ts (hunk 4 above)
Comment (NiclasHedam) [reply packages/alfa-css/test/grammars/text-decoration-style.spec.ts]: Yup

**Comment by kasperisager** [comment]
@NiclasHedam Needs a bit of conflict resolution and then it's good to go.

**Pull request merged**